lib/logstash/filters/grok.rb

```diff
@@ -331,6 +326,24 @@ def close
 
     private
 
+    # The default pattern paths, depending on environment.
+    def patterns_path
+      patterns_path = []
+      case (ecs_compatibility rescue :disabled) # TODO
+      when :disabled
+        patterns_path << LogStash::Patterns::Core.path #(:legacy)
+      when :v1
+        patterns_path << LogStash::Patterns::Core.path(:v1)
+      else
+        fail(NotImplementedError, "ECS #{ecs_compatibility} is not supported by this plugin.")
+      end
+      # allow plugin to be instantiated outside the LS environment (in tests)
+      if defined? LogStash::Environment.pattern_path
```

Comment:
🤔 Is this safeguard needed? A lot of Logstash is needed for this class to even get defined since we inherit `LogStash::Filters::Base` and also `require "logstash/environment"`.

Reply:
runnings specs if you just `require "logstash/filters/grok"` you will need to patch that in first.
like the patterns gem does: https://github.com/logstash-plugins/logstash-patterns-core/blob/v4.1.2/spec/spec_helper.rb#L15-L24
seen it elsewhere as well, it's a bit annoying and counter intuitive to load `logstash/environment` in tests.